godot-ffi/src/global.rs

```diff
@@ -172,7 +186,9 @@ pub enum GlobalLockError<'a, T> {
     WouldBlock,
 
     /// A panic occurred while a lock was held. This excludes initialization errors.
-    Poisoned { circumvent: GlobalGuard<'a, T> },
+    Poisoned {
+        circumvent: Option<GlobalGuard<'a, T>>,
+    },
```

Comment:
This weakens the API type safety towards the user -- I don't think there's a case where this can be `None`?

Reply:
i dont think so, but it is a bit hard to be 100% confident about it. The only time it can be `None` is if there is a panic when the `InitState` mutex is locked, but hasn't been set to `Initialized` yet, and also isn't caught by the `catch_unwind`. And while i dont think that can happen, it's rather involved control flow and i havent been able to fully convince myself that it definitely wont happen. 

Another note is that it's also not actually accessed anywhere. the only place the `Poisoned` variant is used it also just discards the `circumvent` value.

Another option than using `new_unchecked()` would be `new().expect()` which makes it not (library) UB to create the `GlobalGuard`, but does throw a panic. However if we are pretty certain that this wont happen, and this error case is fairly rare anyway, then maybe that's just fine?.

Reply:
Generally I'd tend to make it as expressive as possible (i.e. not `Option`), and if our assumptions end up being wrong, treat it as a bug to be fixed.

I think `new().expect()` would be the safe choice here, retrieving poisoned guards shouldn't be the hot path. If it turns out that we limit an actual use case, then we'll have to make a (breaking) API change.